### PR TITLE
ROX-22101: Create gogo proto Timestamps through compatibility layer

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The goal is to remove direct calls to the gogo protobuf layer and have all of them happen through an abstraction layer in order to ease the migration to a new protobuf library.

This PR moves the direct Timestamp instantiations from the gogo protobuf package to the protocompat abstraction layer.

The change is the application through the refactoring tool `eg` of the following template:

```
package refactor

import (
	"github.com/gogo/protobuf/types"
	"github.com/stackrox/rox/pkg/protocompat"
)

func before(seconds int64) *types.Timestamp {
	return &types.Timestamp{Seconds: seconds}
}

func after(seconds int64) *types.Timestamp {
	return protocompat.GetProtoTimestampFromSeconds(seconds)
}
```

PR stack:
- #9148
  - #9156 
  - #9145 
  - #9196 
  - #9198
  - #9201

## Checklist
- [ ] Investigated and inspected CI test results
~~- [ ] Unit test and regression tests added~~
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Rely on CI coverage to ensure no negative impact from the refactor

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/timestamp/microts.go [hunk 1]
```diff
@@ -76,10 +77,7 @@ func (ts MicroTS) Protobuf() *timestamp.Timestamp {
 
 // GogoProtobuf converts this microtimestamp to a (Gogo) protobuf representation.
 func (ts MicroTS) GogoProtobuf() *types.Timestamp {
```
Comment (mtodor) [inline pkg/timestamp/microts.go:79]: As I remember, our goal with the compatibility layer is to remove `gogo` dependency from our code:
```
	"github.com/gogo/protobuf/types"
```

And keep it isolated only in`protocompat` layer to have a single place to change when we switch to another generator. This package still uses `gogo/protobuf/types`.

Is this addressed in some follow-up PR?
Or should we adjust this to add compatibility layer where `GogoProtobuf()` is called.
Theoretically, we should remove `GogoProtobuf()`. But also need to check if that "interface" is used in scanner.

**Review by mtodor — COMMENTED** [review]
There is one comment. It's mostly related to function that actually uses gogoproto types in definition.

File: pkg/timestamp/microts.go (hunk 1 above)
Comment (rhybrillou) [reply pkg/timestamp/microts.go:79]: I plan a follow-up PR where the remaining usage of "github.com/gogo/protobuf" sublibraries out of the proto* packages is removed. This one is not out yet, though.

**Review by mtodor — APPROVED** [review]

**Pull request merged**